spec/ruby/language/constants_spec.rb

```diff
@@ -134,20 +134,6 @@ class << self
       }.should complain(/already initialized constant/)
       ConstantSpecs::ClassB::CS_CONST109.should == :const109_2
     end
-
-    it "evaluates the right hand side before evaluating a constant path" do
-      mod = Module.new
-
-      mod.module_eval <<-EOC
-        ConstantSpecsRHS::B = begin
-          module ConstantSpecsRHS; end
-
-          "hello"
-        end
-      EOC
-
-      mod::ConstantSpecsRHS::B.should == 'hello'
-    end
```

Comment:
Could you add a spec describing the new behavior?

Also, we should keep this spec for Ruby < 3.1 to not decrease coverage.
It seems likely this change could introduce some incompatibility, and it seems unlikely to be backported (if it is, the version guard can be changed), so keeping the spec (with a version guard) makes a lot of sense.

Reply:
This spec describes an unintentional, buggy behavior.  It is worse to have a wrong spec than no spec at all. While I agree this bugfix is unlikely to be backported, ultimately such a choice is left to the branch maintainer. If TruffleRuby/JRuby want to target Ruby 2.6/2.7/3.0 and want to make this change earlier, I don't think we should have a spec preventing it.  Keeping the spec for previous versions codifies wrong behavior.

I don't have a problem with a spec being added for the correct behavior, but it is not something I plan to work on myself.

Reply:
I replied on https://bugs.ruby-lang.org/issues/15928#note-7, it seems best to keep the conversion in one place (sorry about that)

Reply:
As far as it is only for the coverage, it should be `ruby_bug` instead of `ruby_version_is` at least.

Reply:
`ruby_bug` is useless for other Ruby implementation, the previous behavior is the behavior of every Ruby < 3.2 and it sounds very unlikely to be backport, hence `ruby_version_is ""..."3.2" do` is correct.
Concretely, this spec is useful for compatibility for Ruby < 3.2 for JRuby, TruffleRuby, etc.
User code might depend on it.